packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/AmplifyDataStorePlugin.kt

```diff
@@ -15,37 +15,161 @@
+import java.util.Date
+import java.util.UUID
 
 /** AmplifyDataStorePlugin */
-class AmplifyDataStorePlugin: FlutterPlugin, MethodCallHandler {
-  /// The MethodChannel that will the communication between Flutter and native Android
-  ///
-  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
-  /// when the Flutter Engine is detached from the Activity
-  private lateinit var channel : MethodChannel
-
-  override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
-    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "amplify_datastore")
-    channel.setMethodCallHandler(this)
-  }
-
-  override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
-    if (call.method == "getPlatformVersion") {
-      result.success("Android ${android.os.Build.VERSION.RELEASE}")
-    } else {
-      result.notImplemented()
+class AmplifyDataStorePlugin : FlutterPlugin, MethodCallHandler {
+
+    private lateinit var channel: MethodChannel
+    private val LOG = Amplify.Logging.forNamespace("amplify:flutter:datastore")
+
+    override fun onAttachedToEngine(
+            @NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
+        channel = MethodChannel(flutterPluginBinding.binaryMessenger,
+                                "com.amazonaws.amplify/datastore")
+        channel.setMethodCallHandler(this)
+        LOG.info("Added DataStore plugin")
+    }
+
+    override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
+        var data: HashMap<String, Any> = HashMap()
+        try {
+            data = checkArguments(call.arguments) as HashMap<String, Any>
+        } catch (e: Exception) {
+            prepareError(result, e, FlutterDataStoreFailureMessage.ERROR_CASTING_INPUT_IN_PLATFORM_CODE.toString())
+        }
+        when (call.method) {
+            "query" -> onQuery(result, data)
+            "configure" -> onConfigure(result, data)
+            else -> result.notImplemented()
+        }
     }
-  }
 
-  override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
-    channel.setMethodCallHandler(null)
-  }
+    private fun onConfigure(flutterResult: Result, request: HashMap<String, Any>) {
+        var modelSchemas: List<Map<String, Any>>? = null
+        if (request.containsKey("modelSchemas") && request["modelSchemas"] is List<*>) {
+            modelSchemas = request["modelSchemas"].safeCastToList()
+        } else {
+            prepareError(flutterResult,
+                         Exception(FlutterDataStoreFailureMessage.AMPLIFY_QUERY_REQUEST_MALFORMED.toString()),
+                         FlutterDataStoreFailureMessage.ERROR_CASTING_INPUT_IN_PLATFORM_CODE.toString())
+        }
+
+        val modelProvider = FlutterModelProvider.instance
+        val flutterModelSchemaList =
+                modelSchemas!!.map { modelSchema -> FlutterModelSchema(modelSchema) }
```

Comment:
This was because of casting from serialized data. For now I forced the serialization. the configure implementation is not final here and is a placeholder for us to test.